packages/nextjs/components/BatchStatusIcon.tsx

```diff
@@ -0,0 +1,96 @@
+import { FaCheckCircle, FaTimesCircle } from "react-icons/fa";
+import { useScaffoldReadContract } from "~~/hooks/scaffold-eth";
+
+export const BatchStatusIcon = ({ userAddress }: { userAddress: string | undefined }): React.ReactNode => {
+  const { data: allowList } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "allowList",
+    args: [userAddress],
+  });
+
+  const { data: yourContractAddress } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "yourContractAddress",
+    args: [userAddress],
+  });
+
+  function isInBatch13() {
+    return allowList;
+  }
+  function isCheckedIn() {
+    return yourContractAddress == "0x0000000000000000000000000000000000000000" ? false : true;
+  }
```

Comment:
When using React, developers usually prefer to rely on variables that directly reflect the state of the component because it's easier to read. 
Also, the name `isInBatch13` implies it is a boolean, so I would "transform" `allowList` into a boolean using the double-negative hack (`!!`).
And it's always good to use `===` to check variables with the exact type. This way, the browser will avoid making conversions.
See my suggestion:
```suggestion
  const isInBatch13 = !!allowList;
  const isCheckedIn = yourContractAddress === "0x0000000000000000000000000000000000000000" ? false : true;
```

Reply:
Thank you very helpful, I didn't know about the !! and it's more concise. I will go with this suggestion.